### ISSUE TITLE
There is no variable named "dependency" - terragrunt 0.58.14 (again, 3rd time)

### ISSUE THREAD

**Issue by rios0rios0** [gruntwork-io/terragrunt#3182]
## Describe the bug

I'm getting the following error when trying to `init`, `plan`, or `apply` changes with Terragrunt:

```
INFO[0000] Downloading Terraform configurations from file:///home/rios0rios0/Development/dev.azure.com/REDACTED/REDACTED/REDACTED-template-core into /home/rios0rios0/Development/dev.azure.com/REDACTED/REDACTED/REDACTED-template-core/environments/shared/.terragrunt-cache/nyXe9Z9PiquBT8DUAoSCncjSvGk/h0XYeJQNSv4_0IgqKjiLUu2NBkU  prefix=[/home/rios0rios0/Development/dev.azure.com/REDACTED/REDACTED/REDACTED-template-core/environments/shared] 
WARN[0000] Error reading file /home/rios0rios0/Development/dev.azure.com/REDACTED/REDACTED/shared: open /home/rios0rios0/Development/dev.azure.com/REDACTED/REDACTED/shared: no such file or directory
WARN[0000] Error reading partial config for dependency base_platform: open /home/rios0rios0/Development/dev.azure.com/REDACTED/REDACTED/shared: no such file or directory
ERRO[0000] Error: Unknown variable

ERRO[0000]   on /home/rios0rios0/Development/dev.azure.com/REDACTED/REDACTED/REDACTED-template-core/environments/terragrunt.hcl line 19, in remote_state:
ERRO[0000]   19:     resource_group_name  = dependency.shared.outputs.resource_group_name
ERRO[0000]
ERRO[0000] There is no variable named "dependency".

ERRO[0000] Error: Unknown variable

ERRO[0000]   on /home/rios0rios0/Development/dev.azure.com/REDACTED/REDACTED/REDACTED-template-core/environments/terragrunt.hcl line 20, in remote_state:
ERRO[0000]   20:     storage_account_name = dependency.shared.outputs.storage_account_name
ERRO[0000]
ERRO[0000] There is no variable named "dependency".

ERRO[0000] Error: Unknown variable

ERRO[0000]   on /home/rios0rios0/Development/dev.azure.com/REDACTED/REDACTED/REDACTED-template-core/environments/terragrunt.hcl line 21, in remote_state:
ERRO[0000]   21:     container_name       = dependency.shared.outputs.storage_container_name
ERRO[0000]
ERRO[0000] There is no variable named "dependency".

INFO[0000] Downloading Terraform configurations from file:///home/rios0rios0/Development/dev.azure.com/REDACTED/REDACTED/REDACTED-template-core into /home/rios0rios0/Development/dev.azure.com/REDACTED
Security/REDACTED/REDACTED-template-core/environments/base_platform/REDACTEDlab/.terragrunt-cache/mgJLXhz6SP8P90oxMptWFr188kA/h0XYeJQNSv4_0IgqKjiLUu2NBkU  prefix=[/home/rios0rios0/Development/dev.azure.com/REDACTED/REDACTED/REDACTED-template-core/environments/base_platform/REDACTEDlab]
INFO[0002] Downloading Terraform configurations from file:///home/rios0rios0/Development/dev.azure.com/REDACTED/REDACTED/REDACTED-template-core into /home/rios0rios0/Development/dev.azure.com/REDACTED/REDACTED/REDACTED-template-core/environments/apps_platform/REDACTEDlab/.terragrunt-cache/5DlQdH6Fu0HXpdX9TZRfBZ5RXV4/h0XYeJQNSv4_0IgqKjiLUu2NBkU
INFO[0005] Running [terragrunt apply] in /home/rios0rios0/Development/dev.azure.com/REDACTED/REDACTED/REDACTED-template-core/environments/apps_platform/REDACTEDlab 
WARN[0000] Error reading file /home/rios0rios0/Development/dev.azure.com/REDACTED/REDACTED/shared: open /home/rios0rios0/Development/dev.azure.com/REDACTED/REDACTED/shared: no such file or directory
WARN[0000] Error reading partial config for dependency base_platform: open /home/rios0rios0/Development/dev.azure.com/REDACTED/REDACTED/shared: no such file or directory
ERRO[0000] Error: Unknown variable

ERRO[0000]   on /home/rios0rios0/Development/dev.azure.com/REDACTED/REDACTED/REDACTED-template-core/environments/terragrunt.hcl line 19, in remote_state:
ERRO[0000]   19:     resource_group_name  = dependency.shared.outputs.resource_group_name
ERRO[0000]
ERRO[0000] There is no variable named "dependency".

ERRO[0000] Error: Unknown variable

ERRO[0000]   on /home/rios0rios0/Development/dev.azure.com/REDACTED/REDACTED/REDACTED-template-core/environments/terragrunt.hcl line 20, in remote_state:
ERRO[0000]   20:     storage_account_name = dependency.shared.outputs.storage_account_name
ERRO[0000]
ERRO[0000] There is no variable named "dependency".

ERRO[0000] Error: Unknown variable

ERRO[0000]   on /home/rios0rios0/Development/dev.azure.com/REDACTED/REDACTED/REDACTED-template-core/environments/terragrunt.hcl line 21, in remote_state:
ERRO[0000]   21:     container_name       = dependency.shared.outputs.storage_container_name
ERRO[0000]
ERRO[0000] There is no variable named "dependency".
```

## Steps To Reproduce

You just need to have a "terragrunt.hcl" with 2 "dependency" block, and that's it.

```hcl
generate "providers" {
  path      = "providers.default.tf"
  if_exists = "overwrite_terragrunt"
  contents  = <<EOF
        provider "azurerm" {
            features {}
        }
EOF
}

dependency "1" {
  config_path = "../../1"
}


dependency "2" {
  config_path = "../../2"
}

remote_state {
  backend = "azurerm"
  config = {
    resource_group_name  = dependency.1.outputs.resource_group_name
    storage_account_name = dependency.1.outputs.storage_account_name
    container_name       = dependency.1.outputs.storage_container_name
    key                  = "${basename(dirname(get_terragrunt_dir()))}/${basename(get_terragrunt_dir())}.tfstate"
  }

  generate = {
    path      = "backend.tf"
    if_exists = "overwrite_terragrunt"
  }
}

terraform {
  source = "doesn't matter"
}

inputs = {
  doesn't matter 
}
```

## Expected behavior

No errors or warnings should be thrown, as with Terragrunt 0.55.3 and earlier.

## Nice to haves

- [x] Terminal output
- [ ] Screenshots

## Versions

- Terragrunt version: 0.58.14
- OpenTofu/Terraform version: 1.8.4
- Environment details (Ubuntu 20.04, Windows 10, etc.): linux_amd64

## Additional context

The same error happened more than 3 times in Terragrunt history my friends. Please, let's pay attention to this:
https://github.com/gruntwork-io/terragrunt/issues/2955
https://github.com/gruntwork-io/terragrunt/issues/2956

**Comment by levkohimins** [comment]
Hi @rios0rios0,

I couldn't reproduce the issue, the example I tested: https://github.com/levkohimins/terragrunt-issue-3182

Terragrunt version: [v0.59.1](https://github.com/gruntwork-io/terragrunt/releases/tag/v0.59.1)

**Comment by levkohimins** [comment]
It seems the problem is solved. So I'm closing the issue. If not, feel free to open it again.

**Comment by rios0rios0** [comment]
> Hi @rios0rios0,
> 
> I couldn't reproduce the issue, the example I tested: https://github.com/levkohimins/terragrunt-issue-3182
> 
> Terragrunt version: [v0.59.1](https://github.com/gruntwork-io/terragrunt/releases/tag/v0.59.1)
> 
> ```shell
> run-all apply -auto-approve --terragrunt-log-level debug --terragrunt-working-dir /path/to/root/dir
> ```

Hi @levkohimins.
Sorry for my delay in the answer. Well, you should have 2 dependencies in just 1 ".hcl" file. Not just one. Both HCL files should have content. Could you try this approach?

**Comment by rios0rios0** [comment]
It was solved in the newest version: v0.66.3
